src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -117,8 +117,12 @@ private JSONObject getSteps(String nodeId) {
     public void getAllSteps(StaplerRequest2 req, StaplerResponse2 rsp) throws IOException, ServletException {
         run.checkPermission(Item.READ);
 
+        // Look up completed state before computing steps.
+        boolean runIsComplete = !PipelineState.of(run).isInProgress();
+
         PipelineStepList steps = stepApi.getAllSteps();
         JSONObject json = JSONObject.fromObject(steps, jsonConfig);
+        json.element("runIsComplete", runIsComplete, jsonConfig);
```

Comment:
I'll try and have a proper look tomorrow but rather than just adding this to the JSON object I think it would be better to add it to the class that the JSON object is created from (`PipelineStepList`).

Reply:
:+1: Done in 0c1d96ef1c3de63566ea262377bf75b6e2061ed9.